### ISSUE TITLE
Helper to optimize JSON-LD metadata report for Google dataset search ingestion

### ISSUE THREAD

**Issue by mih** [datalad/datalad-metalad#24]
It looks as if we can already get quite far in terms of getting schema.org metadata out of DataLad and into Google's dataset search. Now we need to make it more convenient to get practical benefits out of this. Few pointers to guide next steps:

- demo dataset that is built from two RSS feeds: https://github.com/datalad-datasets/longnow-podcasts
- small script to turn the RSS feed into schema.org metadata, structured for access via the `custom` metadata extractor: https://github.com/datalad-datasets/longnow-podcasts/blob/master/.datalad/maint/update_metadata_from_feed.py
- helper to format a README.md based on DataLad's aggregated metadata. This is trivial to mark up with embedded [microdata](https://schema.org/docs/gs.html#microdata_embedded). Either following the example of @chrisgorgo (https://github.com/googlecreativelab/quickdraw-dataset/pull/32), or with a non-table-like inline markup ([PoC](https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fgist.github.com%2Fmih%2F3d6830433573524c3e6576983cbded70)). Both work equally well.

Beyond this custom approach that would need adjustments and coding for each dataset, a more generic solution would be preferable. The [output](https://github.com/datalad/datalad-metalad/files/3392053/meta_dump.txt) of the generic call:

```
datalad --format json meta-dump --reporton jsonld | jq '.metadata'
```

leaves the Google testing tool [in pain](https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fgithub.com%2Fdatalad%2Fdatalad-metalad%2Ffiles%2F3392053%2Fmeta_dump.txt). But the majority of the issues seem fixable:

- [ ] undefined term `contentbytesize` (unclear if there is a better existing one)
- [ ] Google doesn't like `hasContributor` as a `Dataset` property 
- [ ] a `DigitalDocument` cannot have a `distribution`. But for DataLad a plain URL isn't enough, so maybe we have to strip that from an export to make Google happy
- [ ] Google does not recognize PROV-type `agent` (or definition is missing, needs check)

**Comment by adswa** [comment]
I've only quickly looked into terms of schema.org, but there is [contentSize](https://schema.org/contentSize) as a Property: File size in (mega/kilo) bytes.
Might that be an alternative to ``contentbytesize``?

**Comment by mih** [comment]
True. However, it requires some kind of unit specification, which is a bit of a burden for any further processing. Related is the general issue of representing quantitative measures: https://github.com/datalad/datalad-metalad/issues/8

**Comment by chrisgorgo** [comment]
Great to see this moving forward! 

I would recommend starting small - just with `name`, `description`, and `sameAs` (pointing to canonical URL - helps with deduplicating forks) for the whole dataset without listing all individual files or relationships between datasets. That alone should get the datasets indexed. Then tackling `license` and `measuredVariables` would be probably most useful from the findability point of view.

Some additional resources:
- [all schema.org/Dataset fields](https://schema.org/Dataset)
- [subset of schema.org/Dataset fields that currently are or will in the near future be used by Dataset Search](https://developers.google.com/search/docs/data-types/dataset)

**Comment by mih** [comment]
Thx @chrisgorgo ! I would expect more movement in the near future.

Quick Q: how does the Google react to additional information? Is it friendly ignorance or rather scared refusal? ;-)

We are moving towards more comprehensive descriptions and models quickly. Q is whether to keep something simple, targeted, separately on the side, or whether to put all eggs in one basket?

This one is a non-science example, but we can now pull information across multiple NIDM flavours, down to the file level.

**Comment by chrisgorgo** [comment]
Friendly ignorance :)

Have a look at metadata on zenodo.org, figshare.com, Harvard Dataverse,
NeuroVault.org, kaggle.com and how the corresponding datasets are rendered
on g.co/DatasetSearch. This should give you better overview of how things
can look when the work is done.
